app/src/main/java/com/mstudio/superheroarch/presentation/DetailsViewModel.kt

```
@@ -23,18 +25,46 @@ class DetailsViewModel(
+            view.showError("Failed to load episode details: ${e.message}")
+        }
+    }
+
+
+    fun extractSeasonAndEpisode(firstEpisode: String): Pair<Int, Int>? {
+        return if (firstEpisode.isNotEmpty()) {
+            try {
+                val season = firstEpisode.substring(1, 3).toInt()
+                val episode = firstEpisode.substring(4, 6).toInt()
+                Pair(season, episode)
+            } catch (e: Exception) {
```

detekt

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)